federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -931,31 +896,10 @@ func (s *ServiceController) processServiceUpdate(cachedService *cachedService, s
 // we should retry in that Duration.
 func (s *ServiceController) processServiceDeletion(key string) (error, time.Duration) {
 	glog.V(2).Infof("Process service deletion for %v", key)
-	cachedService, ok := s.serviceCache.get(key)
-	if !ok {
-		return fmt.Errorf("Service %s not in cache even though the watcher thought it was. Ignoring the deletion.", key), doNotRetry
-	}
-	service := cachedService.lastState
-	cachedService.rwlock.Lock()
-	defer cachedService.rwlock.Unlock()
-	s.eventRecorder.Event(service, v1.EventTypeNormal, "DeletingDNSRecord", "Deleting DNS Records")
-	// TODO should we delete dns info here or wait for endpoint changes? prefer here
-	// or we do nothing for service deletion
+	// TODO: Should uncomment this?
+	// We should delete the load balancer when service is deleted
+	// Look at the same method in kube service controller.
 	//err := s.dns.balancer.EnsureLoadBalancerDeleted(service)
```

Comment:
umm. Why not? kube service controller calls this.
Leaving this as is. Can remove it in a separate PR if we want to.